packages/named/index.ts

```diff
@@ -6,7 +6,7 @@ import compose, { Stamp } from '@stamp/compose';
 const Named = compose({
   staticProperties: {
     setName(this: Stamp, name: string): Stamp {
-      return (this?.compose ? this : Named).compose({
+      return (this ? this : Named).compose({
```

Comment:
I know TS kinda protects us from this mistake. But only 50% of JS devs use TS. Please, return the logic or change to this:
```js
return (isStamp(this) ? this : Named).compose({
```

Reply:
The isStamp angle is what I'll try or tsc will complain

Reply:
isStamp does indeed solve the issue. Thanks!